### PR TITLE
Add option to update helm dependencies

### PR DESCRIPTION
**Pull request by hendrikmaus**
#46 

- harden the test cases for render module by wrapping the struct creation into utility functions. this allows for easier easier extension of the structs under test as one only has to edit the helper functions when a new field is added, rather than *n* test cases which create a struct.
- factor the helm command execution into a resuable function
- create an ordered hashmap (IndexMap) to collect pre-commands to run *once* before rendering the deployments.
- changed the log output to be more readable: 
image

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -61,6 +61,10 @@ pub struct RenderCmdOpts {
     /// Print rendered manifests to stdout
     #[structopt(short, long)]
     stdout: bool,
+
+    /// Run `helm dependencies update` before rendering deployments
+    #[structopt(short, long)]
+    update_dependencies: bool,
```
Comment (Kenec) [inline src/main.rs:67]: Is there a case where we don't want to update the dependencies? I reckon that is why this is set to a `bool`

File: src/render_cmd.rs [hunk 2]
```diff
@@ -168,67 +188,98 @@ impl RenderCmd {
 
     /// Execute the commands in the given plan
     fn exec_plan(&self, plan: &Plan) -> anyhow::Result<()> {
-        for (deployment, cmd) in &plan.commands {
-            info!(" - {}", deployment);
+        if !&plan.pre_commands.is_empty() {
```
Comment (Kenec) [inline src/render_cmd.rs:191]: Is the check for empty `pre_commands` better off at the `plan` stage where you are checking if the `update_dependencies` flag is there? ie after here https://github.com/hendrikmaus/helm-templexer/pull/60/files#diff-a374dd3d48a9861daacaafcecde4f50fcf330ea0e490f16cf6359f564f2d5c9fR103

File: src/main.rs (hunk 1 above)
Comment (hendrikmaus) [reply src/main.rs:67]: Yes, by default it is false as this is often not required.

File: src/render_cmd.rs (hunk 2 above)
Comment (hendrikmaus) [reply src/render_cmd.rs:191]: The plan phase populates the `pre_commands` map, and here we check if there is anything to execute.

File: src/main.rs (hunk 1 above)
Comment (hendrikmaus) [reply src/main.rs:67]: It is only useful once you start to render charts which have remote sub-charts as dependencies.

**Review by Kenec — APPROVED** [review]

**Pull request merged**